harmoni_detectors/harmoni_stt/test/test_google_stt.py

```diff
@@ -0,0 +1,129 @@
+        self.feedback = State.INIT
+        self.result = False
+        self.test_file = rospy.get_param("test_google_stt_input")
+        self.audio = self.wav_to_data(self.test_file)
+        rospy.init_node("test_google_stt", log_level=rospy.INFO)
+        self.rate = rospy.Rate(20)
+        self.output_sub = rospy.Subscriber(
+            "/harmoni/detecting/stt/default", String, self._detecting_callback
+        )
+
+        # provide mock microphone
+        self.audio_pub = rospy.Publisher(
+            SensorNameSpace.microphone.value + "default",
+            AudioData,
+            queue_size=10,
+        )
+        rospy.Subscriber(
+            DetectorNameSpace.stt.value + "stt_default",
+            String,
+            self.text_received_callback,
+        )
+
+        # startup stt node
+        self.server = "stt_default"
+        self.client = HarmoniActionClient(self.server)
+        self.client.setup_client(
+            self.server, self._result_callback, self._feedback_callback, wait=True
+        )
+        rospy.loginfo("TestGoogleStt: Turning ON stt server")
+        self.client.send_goal(action_goal=ActionType.ON, optional_data="Setup", wait=False)
+        
+        time.sleep(3)
+        # self.client.send_goal(
+        #                 action_goal=ActionType.REQUEST.value,
+        #     wait=False
+        # )
+        rospy.loginfo("TestGoogleStt: Started up. waiting for google stt startup")
+
+        # wait for start state
+        #while not rospy.is_shutdown() and self.feedback != State.START:
+        #    self.rate.sleep()
+
+        rospy.loginfo("TestGoogleStt: publishing audio")
+        
+        self.audio_pub.publish(self.audio)
+        self.audio_pub.publish(self.audio[:14000])
+
+        rospy.loginfo(
+            f"TestGoogleStt: audio subscribed to by #{self.output_sub.get_num_connections()} connections."
+        )
+
+        time.sleep(5)
+
+    def _feedback_callback(self, data):
+        rospy.loginfo(f"TestGoogleStt: Feedback: {data}")
+        self.feedback = data["state"]
+
+    def _status_callback(self, data):
+        rospy.loginfo(f"TestGoogleStt: Status: {data}")
+        self.result = True
+
+    def _result_callback(self, data):
+        rospy.loginfo(f"TestGoogleStt: Result: {data}")
+        self.result = True
+
+    def text_received_callback(self, data):
+        rospy.loginfo(f"TestGoogleStt: Text back: {data}")
+        self.result = True
+
+    def _detecting_callback(self, data):
+        rospy.loginfo(f"TestGoogleStt: Detecting: {data}")
+        self.result = True
+
+
+class TestGoogleStt_Valid(TestGoogleStt_Common):
+    def test_IO(self):
+        rospy.loginfo(
+            "TestGoogleStt[TEST]: basic IO test to ensure data "
+            + "('hello' audio) is received and responded to. Waiting for transcription..."
+        )
+        while not rospy.is_shutdown() and not self.result:
+            self.rate.sleep()
+        assert self.result == True
+
```

Comment:
I would add another test to verify the text received is the same as what is expected.

Reply:
Thank you for your suggestion! I was a bit stuck. 
However, I have just found out that the code the code here for the stt service is not the same as the one in feature/edison-smarthome. I changed it a while ago because it was giving me "ERROR: Broken Pipe" when using the method transcribe_stream_request. So I followed the instruction on the google stt page to create a different method for a continuous stream from the mic.

TLDR: I cannot add another test since this stt service is not working (for me at least).
I can add a test if I change the stt service file to the one in feature/edison-smarthome, where there is a different method for handling a stream.